VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs

```diff
@@ -162,5 +170,53 @@ protected virtual void AddParentOrganizations(IndexDocument document, ICollectio
             document.AddFilterableValues("ParentOrganizations", nonEmptyValues);
             document.AddFilterableValue("HasParentOrganizations", nonEmptyValues?.Any() ?? false);
         }
+
+        protected virtual void IndexDynamicProperties(IndexDocument document, ICollection<DynamicObjectProperty> properties)
+        {
+            foreach (var property in properties)
+            {
+                var propertyName = property.Name?.ToLowerInvariant();
+                if (!string.IsNullOrEmpty(propertyName))
+                {
+                    var isCollection = property?.IsDictionary == true || property?.IsArray == true;
+                    IList<object> values = new List<object>();
+
+                    switch (property.ValueType)
+                    {
+                        case DynamicPropertyValueType.Boolean:
+                        case DynamicPropertyValueType.DateTime:
+                        case DynamicPropertyValueType.Integer:
+                        case DynamicPropertyValueType.Decimal:
+                            if (property?.IsArray == true)
+                            {
+                                values.AddRange((property.Values as IEnumerable<DynamicPropertyObjectValue>).Select(x => x.Value));
+                                document.Add(new IndexDocumentField(propertyName, values) { IsRetrievable = true, IsFilterable = true, IsCollection = isCollection });
+                            }
+                            else
+                            {
+                                document.Add(new IndexDocumentField(propertyName, property.Values.FirstOrDefault()?.Value) { IsRetrievable = true, IsFilterable = true, IsCollection = isCollection });
+                            }
+
+                            break;
+                        case DynamicPropertyValueType.ShortText:
+                            if (property?.IsDictionary == true)
+                            {
+                                //add all locales in dictionary to searchIndex
+                                values.AddRange((property.Values as IEnumerable<DynamicPropertyObjectValue>).Select(x => x.Value)
+                                                                        .Cast<DynamicPropertyDictionaryItem>()
+                                                                        .Select(x => x.Name));
+                            }
+
+                            else if (property?.IsArray == true)
+                            {
+                                values.AddRange((property.Values as IEnumerable<DynamicPropertyObjectValue>).Select(x => x.Value));
```

Comment:
Remove  `as IEnumerable<DynamicPropertyObjectValue>`
To avoid this and previous defects in future, install ReSharper.